PGM/src/main/java/tc/oc/pgm/shop/shops/BlockShop.java

```diff
@@ -0,0 +1,57 @@
+package tc.oc.pgm.shop.shops;
+
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.EventPriority;
+import org.bukkit.event.Listener;
+import org.bukkit.event.player.PlayerInteractEvent;
+import org.bukkit.util.Vector;
+import tc.oc.pgm.events.ListenerScope;
+import tc.oc.pgm.filters.Filter;
+import tc.oc.pgm.match.Match;
+import tc.oc.pgm.match.MatchPlayer;
+import tc.oc.pgm.match.MatchScope;
+import tc.oc.pgm.shop.PurchaseTracker;
+import tc.oc.pgm.shop.Shop;
+import tc.oc.pgm.shop.purchasable.PurchasableSet;
+
+import java.util.Set;
+
+/**
+ * Represents a shop activated by right clicking on a {@link org.bukkit.block.Block}.
+ */
+@ListenerScope(MatchScope.RUNNING)
+public class BlockShop extends Shop implements Listener {
```

Comment:
I see no reason not to include this as a shop, let map makers decide if they want to use an entity or a block to represent the shop (it could be a sign, for example)